stdlib/LinearAlgebra/src/cholesky.jl

```diff
@@ -835,3 +835,21 @@ then `CC = cholesky(C.U'C.U - v*v')` but the computation of `CC` only uses
 `O(n^2)` operations.
 """
 lowrankdowndate(C::Cholesky, v::AbstractVector) = lowrankdowndate!(copy(C), copy(v))
+
+
+"""
+    diag(C::Cholesky) -> Vector
+
+return the diag(Array(C)).
+"""
+function diag(C::Cholesky{T}) where {T}
+    N = size(C, 1)
+    z = zeros(T, N)
```

Comment:
z = Vector{T}(undef, N) should be more performant here

Reply:
true, I chose this because I needed to initialize the vector with zeros so that the sum made sense. But I have updated this so the element gets updated when necessary.